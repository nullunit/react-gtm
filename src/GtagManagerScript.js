import PropTypes from 'prop-types';
import React from 'react';

export default class GtagManagerScript extends React.Component {

    static propTypes = {
        'children': PropTypes.node.isRequired,
        'id': PropTypes.string.isRequired
    };

    componentDidMount() {

        const { id } = this.props;

        //place the tag manager script in the page
        //(taken from Google Tag Manager console)
        (
            function(w, d, s, l, i) {
                w[l] = w[l] || [];
                w[l].push({ 'gtm.start': new Date().getTime(), 'event': 'gtm.js' });
                const [ f ] = d.getElementsByTagName(s);
                const j = d.createElement(s);
                const dl = l === 'dataLayer' ? '' : `&l=${l}`;
                j.async = true;
                j.src = `https://www.googletagmanager.com/gtm.js?id=${i}${dl}`;
                f.parentNode.insertBefore(j, f);
            }(window, document, 'script', 'dataLayer', id));
    }

    render() {
        const { children } = this.props;
        return React.Children.only(children);
    }

}
