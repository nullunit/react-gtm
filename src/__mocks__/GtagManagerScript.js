import React from 'react';

/**
 * Mock skips adding Google Tag Manager script to page as well as rendering child components.
 */
export default class GtagManagerScript extends React.Component {

    render() {
        return <h1>mock render Google Tag Manager script to page</h1>;
    }

}
