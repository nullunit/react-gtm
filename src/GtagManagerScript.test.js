import GtagManagerScript from './GtagManagerScript';
import React from 'react';
import renderer from 'react-test-renderer';

const GTM_ID = 'GTM-123456';

describe('GtagManagerScript tests', () => {

    it('Append Google Tag Manager script to load Google Analytics', () => {

        const script = '<script type="text/javascript" src="/static/js/bundle.js"></script>';
        const expectedGtagScript = `<script src="https://www.googletagmanager.com/gtm.js?id=${GTM_ID}"></script>${script}`;

        document.body.innerHTML =
            '<html>' +
            `  <body>${script}</body>` +
            '</html>';

        const props = {
            'id': GTM_ID,
            'useTagManager': true
        };

        //test render snapshot
        const json = renderer
            .create(<GtagManagerScript { ...props }><h1>mock child</h1></GtagManagerScript>)
            .toJSON();

        expect(json).toMatchSnapshot();
        expect(expectedGtagScript).toEqual(document.body.innerHTML.trim());
    });

});
