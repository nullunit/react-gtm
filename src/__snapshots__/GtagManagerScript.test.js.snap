// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GtagManagerScript tests Append Google Tag Manager script to load Google Analytics 1`] = `
<h1>
  mock child
</h1>
`;
